crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/prosecution_concluded/service/ProsecutionConcludedService.java

```diff
@@ -67,12 +69,15 @@ public void executeCCOutCome(ProsecutionConcluded prosecutionConcluded, WQHearin
     }
 
     private void processOutcome(ProsecutionConcluded prosecutionConcluded, WQHearingDTO wqHearingDTO, List<OffenceSummary> trialOffences) {
-        prosecutionConcludedValidator.validateOuCode(wqHearingDTO.getOuCourtLocation());
+        String crownCourtCode = prosecutionConcludedValidator.validateOuCode(wqHearingDTO.getOuCourtLocation());
         String calculatedOutcome = calculateOutcomeHelper.calculate(trialOffences);
         log.info("calculated outcome is {} for this maat-id {}", calculatedOutcome, prosecutionConcluded.getMaatId());
 
-        ConcludedDTO concludedDTO = caseConclusionDTOBuilder.build(prosecutionConcluded, wqHearingDTO, calculatedOutcome);
+        ConcludedDTO concludedDTO = caseConclusionDTOBuilder.build(prosecutionConcluded, wqHearingDTO, calculatedOutcome, crownCourtCode);
+        RepOrderDTO repOrderDTO = courtDataAPIService.getRepOrder(concludedDTO.getProsecutionConcluded().getMaatId());
+        List<RepOrderCCOutcomeDTO> repOrderOutcomeDtoList = courtDataAPIService.getRepOrderCCOutcomeByRepId(concludedDTO.getProsecutionConcluded().getMaatId());
 
-        prosecutionConcludedImpl.execute(concludedDTO);
+        prosecutionConcludedValidator.validateMagsCourtOutcomeExists(repOrderDTO, repOrderOutcomeDtoList);
```

Comment:
This check can be simplified and should not require the CC outcome list (see other comment), therefore, I think we should move the validation logic to the top level execute function - we could also group them into a single validate method that assesses the mags outcome and court location code.

Reply:
Will refactor and move up 👍

Reply:
Actually, thinking about this again, we can't group this into the same method as all of the other validation logic unless we are willing to potentially do some work which might not be needed, by which I mean get the _RepOrderDTO_ to obtain the mags court outcome. What do you think?

Reply:
I have a few ideas for how we might get around this without duplicating the call to retrieve the repOrder in the future (caching, adding it as a field in the `ProsecutionConcluded` class), but let's leave that for now